plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/storage/TestTrinoHudiStorage.java

```diff
@@ -293,6 +293,44 @@ private static void validatePathInfoList(
         }
     }
 
+    @Test
+    public void testListEmptyDirectory()
+            throws IOException
+    {
+        HoodieStorage storage = getStorage();
+
+        // Create an empty directory
+        StoragePath emptyDir = new StoragePath(getTempDir(), "empty_directory");
+        assertThat(storage.createDirectory(emptyDir)).isTrue();
+        assertThat(storage.exists(emptyDir)).isTrue();
+
+        // List the empty directory - should return empty list, not throw exception
+        List<StoragePathInfo> entries = storage.listDirectEntries(emptyDir);
+        assertThat(entries).isEmpty();
+
+        List<StoragePathInfo> files = storage.listFiles(emptyDir);
+        assertThat(files).isEmpty();
+    }
+
+    @Test
+    public void testListNonExistentDirectory()
```

Comment:
Ideally the test should verify based on hierarchical and non-hierarchical filesystem, is it hard to write or ?